packages/flutter/lib/src/material/carousel.dart

```diff
@@ -472,6 +509,7 @@ class _CarouselViewState extends State<CarouselView> {
   bool get _consumeMaxWeight => widget.consumeMaxWeight;
   CarouselController? _internalController;
   CarouselController get _controller => widget.controller ?? _internalController!;
+  late int _lastReportedIndex;
```

Comment:
Maybe give it a more clear name to avoid confusion because we have another concept of "initialItem".
```suggestion
  late int _lastReportedLeadingIndex;
```